### ISSUE TITLE
generated models across loaded packages causes error merging type systems

### ISSUE THREAD

**Issue by StevenACoffman** [99designs/gqlgen#1762]
### What happened?
If models are generated in a package that has already been loaded, and
that package refers to another package that has already been loaded, we
can find ourselves in a position where it appears that a GQL `union` is
not satisfied.

https://github.com/99designs/gqlgen/pull/1020 introduces the
code that fails in this scenario. This was supposed to be addressed in https://github.com/99designs/gqlgen/pull/1491 by @wendorf
but per @ipfans https://github.com/99designs/gqlgen/pull/1491#issuecomment-1012046943 it appears to still be occurring.

This repository currently reproduces the issue reliably:
https://github.com/ipfans/gqlgen-error

and is currently getting this error:
```
merging type systems failed: unable to bind to interface: xxx/src/business/entity.A does not satisfy the interface gitlab.com/hookactions/gqlgen-relay/relay.Node
```

**Comment by ipfans** [comment]
I am trying to debug this bug, then I found `cfg.ReloadAllPackages()` is never called before error. After add this, I think everything works now. I will be submit a PR soon.

**Comment by frederikhors** [comment]
Should we release a patch version with all these news?

**Comment by StevenACoffman** [comment]
@frederikhors I released `v0.15.0` right after this merged as this was the final bug fix I was waiting for to land. We're using this same version in production at scale at Khan, and I feel like it's been very solid.

**Comment by frederikhors** [comment]
Yeah it is solid. gqlgen is amazing. Really makes me thankful for life.

Can we open the discussions tab, @StevenACoffman, please, so we can be able to move there issues that are simple questions? (already requested https://github.com/99designs/gqlgen/issues/1722)
